nautobot/users/filters.py

```diff
@@ -32,8 +32,9 @@ class UserFilterSet(BaseFilterSet):
     group_id = django_filters.ModelMultipleChoiceFilter(
         field_name="groups",
         queryset=Group.objects.all(),
-        label="Group",
+        label="Group (ID)",
     )
+    # TODO(timizuo): Migrate ModelMultipleChoiceFilter to NaturalKeyOrPKMultipleChoiceFilter
```

Comment:
IIRC this is blocked because NaturalKeyOrPKMultipleChoiceFilter isn't correctly handling integer inputs?

Reply:
Yup

Reply:
Would you mind clarifying this information in the comment?